ubyssey/views.py

```diff
@@ -442,6 +448,109 @@ def newsletter(self, request):
 
         return render(request, 'objects/newsletter.html', {})
 
+    def get_calendar_events(self, category=None, months=None, start=None, end=None):
+
+        events = Event.objects.all().order_by('start_time')
+        events = events.filter(is_submission=False)
+        events = events.filter(is_published=True)
+
+        today = date.today()
+        # filter start
+        if start is not None:
+            events = events.filter(start_time__gt=start)
+        else:
+            events = events.filter(start_time__gt=today)
+
+        # filter end
+        if end is not None:
+            events = events.filter(end_time__lte=end)
+        else:
+            until_month = today.month + (months if months is not None else 12)
+            until_year = today.year
+            while until_month > 12:
+                until_month -= 12
+                until_year += 1
+            dt_until = today.replace(year=until_year, month=until_month)
+            events = events.filter(end_time__lte=dt_until)
+
+        if category is not None and category != 'all':
+            events = events.filter(category__exact=category)
+
+        HARD_MAX = 100
+        events = events[:HARD_MAX]
+
+        return events
+
+    def organize_events_by_date(self, events):
+        events_by_date = OrderedDict()
+
+        for event in events:
+            start = event.start_time # Zulu
+            start = start.astimezone(timezone('America/Vancouver'))
+            year = start.year
+            month = start.month
```

Comment:
Call this `month_name` for clarity, or just pass it directly into the line below